`error: cannot determine resolution for the macro` itself is not stably triggering on 1.97.1

`src/lib.rs`

```rs
#[cfg(feature = "env-first")]
pub mod env;

// This declaration-before-`env` order is part of the rustc regression.
pub mod interface {
    mod wasm {
        #[allow(unused_imports)]
        use crate::env::{self};

        #[cfg(not(feature = "qualified-env"))]
        include!(concat!(env!("OUT_DIR"), "/wit/dom_export.rs"));

        #[cfg(feature = "qualified-env")]
        include!(concat!(core::env!("OUT_DIR"), "/wit/dom_export.rs"));

        // This child scope survives and drives rustc's `dom_import` suggestion.
        pub mod dom_import {
            include!(concat!(env!("OUT_DIR"), "/wit/dom_import.rs"));
        }

        #[allow(dead_code)]
        struct DomExport;
        impl dom_export::DomExport for DomExport {}
    }

    pub use wasm::*;
}

#[cfg(not(feature = "env-first"))]
pub mod env;
```

`src/env.rs`

```rs
#[derive(Default)]
#[allow(dead_code)]
pub struct BusinessData;
```

`build.rs`

```rs
use std::{env, fs, path::PathBuf};

fn main() {
    let output = PathBuf::from(env::var_os("OUT_DIR").unwrap()).join("wit");
    fs::create_dir_all(&output).unwrap();

    fs::write(
        output.join("dom_export.rs"),
        "pub mod dom_export { pub trait DomExport {} }\n",
    )
    .unwrap();
    fs::write(
        output.join("dom_import.rs"),
        "pub mod dom_import { pub trait DomImport {} }\n",
    )
    .unwrap();
}
```

Building:

```console
$ cargo +1.96.1 build
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.17s

$ cargo +1.97.1 build
error: cannot determine resolution for the macro `env`
  --> src/lib.rs:11:26
   |
11 |         include!(concat!(env!("OUT_DIR"), "/wit/dom_export.rs"));
   |                          ^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error[E0433]: cannot find module or crate `dom_export` in this scope
  --> src/lib.rs:23:14
   |
23 |         impl dom_export::DomExport for DomExport {}
   |              ^^^^^^^^^^ use of unresolved module or unlinked crate `dom_export`
   |
help: there is a crate or module with a similar name
   |
23 -         impl dom_export::DomExport for DomExport {}
23 +         impl dom_import::DomExport for DomExport {}
   |

For more information about this error, try `rustc --explain E0433`.
error: could not compile `xxx` (lib) due to 2 previous errors

$ cargo +1.97.1 build --features env-first
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s

$ cargo +1.97.1 build  --features qualified-env
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.00s
```

If `error: cannot determine resolution for the macro` is an intended feature of 1.97, it should not be affected by the mod declaring order

Workaround: fully qualify to `core::env!` or just don't name the mod `env` and import it

Detail version:

```console
$ rustc +1.97.1 -vV
rustc 1.97.1 (8bab26f4f 2026-07-14)
binary: rustc
commit-hash: 8bab26f4f68e0e26f0bb7960be334d5b520ea452
commit-date: 2026-07-14
host: aarch64-apple-darwin
release: 1.97.1
LLVM version: 22.1.6

$ rustc +1.96.1 -vV
rustc 1.96.1 (31fca3adb 2026-06-26)
binary: rustc
commit-hash: 31fca3adb283cc9dfd56b49cdee9a96eb9c96ffd
commit-date: 2026-06-26
host: aarch64-apple-darwin
release: 1.96.1
LLVM version: 22.1.2
```

Disclosure: this minimal reproduce is generated with GPT-5.6 Sol from diagnose log

--- a/compiler/rustc_resolve/src/ident.rs
+++ b/compiler/rustc_resolve/src/ident.rs
@@ -419,7 +419,7 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         ignore_import: Option<Import<'ra>>,
     ) -> Result<Decl<'ra>, Determinacy> {
         // Make sure `self`, `super` etc produce an error when passed to here.
-        if !matches!(scope_set, ScopeSet::Module(..)) && ident.name.is_path_segment_keyword() {
+        if ident.name.is_path_segment_keyword() {
             return Err(Determinacy::Determined);
         }
 
@@ -1077,10 +1077,6 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                     {
                         let module = self.resolve_crate_root(ident);
                         return Ok(module.self_decl.unwrap());
-                    } else if ident.name == kw::Super {
-                        // FIXME: Implement these with renaming requirements so that e.g.
-                        // `use super;` doesn't work, but `use super as name;` does.
-                        // Fall through here to get an error from `early_resolve_...`.
                     }
                 }
 

--- a/tests/ui/resolve/macro-determinacy-non-module-issue-160195.env_first.stderr
+++ b/tests/ui/resolve/macro-determinacy-non-module-issue-160195.env_first.stderr
@@ -1,0 +1,8 @@
+error: `env!()` takes 1 or 2 arguments
+  --> $DIR/macro-determinacy-non-module-issue-160195.rs:12:22
+   |
+LL |     include!(concat!(env!()));
+   |                      ^^^^^^
+
+error: aborting due to 1 previous error
+

--- a/tests/ui/resolve/macro-determinacy-non-module-issue-160195.env_second.stderr
+++ b/tests/ui/resolve/macro-determinacy-non-module-issue-160195.env_second.stderr
@@ -1,0 +1,8 @@
+error: `env!()` takes 1 or 2 arguments
+  --> $DIR/macro-determinacy-non-module-issue-160195.rs:12:22
+   |
+LL |     include!(concat!(env!()));
+   |                      ^^^^^^
+
+error: aborting due to 1 previous error
+

--- a/tests/ui/resolve/macro-determinacy-non-module-issue-160195.rs
+++ b/tests/ui/resolve/macro-determinacy-non-module-issue-160195.rs
@@ -1,0 +1,21 @@
+//@ revisions: env_first env_second
+
+#[cfg(env_first)]
+pub mod env {
+    #[derive(Default)]
+    pub struct BusinessData;
+}
+
+pub mod interface {
+    use crate::env::{self};
+
+    include!(concat!(env!())); //~ ERROR `env!()` takes 1 or 2 arguments
+}
+
+#[cfg(env_second)]
+pub mod env {
+    #[derive(Default)]
+    pub struct BusinessData;
+}
+
+fn main() {}

